ydb/library/yql/sql/settings/serializer/proto/translation_settings.proto

```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
+
+package NYql.NProto;
+
+option java_package = "com.yandex.yql.proto";
+
+message TTranslationSettings {
```

Comment:
NB: The contents of this .proto are the most import to discuss in this PR.

### Why out of the all fields of the [TTranslationSettings](https://github.com/ydb-platform/ydb/blob/f3a1a6271b5375688cb0608efd644334cc28bbc9/ydb/library/yql/sql/settings/translation_settings.h#L73) I have decided to serialize only:
- PathPrefix
- SyntaxVersion
- AnsiLexer
- Antlr4Parser
- PgParser
- and pragmas?

#### Answer

The scope of the view context capturing in the first release of views is discussed a bit on the [wiki](https://wiki.yandex-team.ru/users/deminds/view-first-release-roadmap/#capture-from-the-very-beginning).

The main reason for choosing only these settings is that they seem to be the most important among those stored in the translation settings struct. Also they [have](https://github.com/ydb-platform/ydb/blob/f3a1a6271b5375688cb0608efd644334cc28bbc9/ydb/library/yql/sql/settings/translation_settings.cpp#L71) dedicated control comments, which somehow makes them more important in my mind.

I tried to implement the following approach:
- capture those fields of the TTranslationSettings struct that affect the compilation of the view query
- inherit those fields of the TTranslationSettings struct that are runtime-affecting from the `SELECT` from the view [context](https://github.com/ydb-platform/ydb-rfc/blob/main/view_context_capture.md#cause)

Here are some examples of the runtime-affecting parts of the TTranslationSettings struct:
- [DefaultCluster](https://github.com/ydb-platform/ydb/blob/f3a1a6271b5375688cb0608efd644334cc28bbc9/ydb/library/yql/sql/settings/translation_settings.h#L96)
  I inherit it from the enclosing parser context:
  - [here](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/library/yql/sql/v1/sql_translation.cpp#L66) for verification of the query during `CREATE VIEW` before storing it in the view
  - [here](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/core/kqp/host/kqp_translate.cpp#L93) during an actual [rewriting](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/core/kqp/provider/yql_kikimr_datasource.cpp#L777) of the read from the view node during `SELECT` from the view
- [ClusterMapping](https://github.com/ydb-platform/ydb/blob/main/ydb/library/yql/sql/settings/translation_settings.h#L78)
  I inherit it from the enclosing parser context:
  - [here](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/library/yql/sql/v1/sql_translation.cpp#L64) for verification of the query text during `CREATE VIEW`
  - [here](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/core/kqp/host/kqp_translate.cpp#L94) for the actual rewriting of a read from a view

Some translation settings are not being captured, because they are irrelevant for views:
- [V0Behavior](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L107C21-L107C31)
  We don't support V0 syntax in views.
- other V0 settings: V0ForceDisable, WarnOnV0, V0WarnAsError
- [DeclaredNamedExprs](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L93C32-L93C50)
  We plan to support named expressions inside view queries with a different approach. This feature is scheduled for the next view release.
- [EndOfQueryCommit](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L99)

Some translation settings are going to be captured in the next release:
- [GUC settings](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L123)
  You can read more on the first and the second releases features on [wiki](https://wiki.yandex-team.ru/users/deminds/view-first-release-roadmap/#view-context-capture-development-plan).
  
Other translation settings are not captured, because I don't know if they are needed for views:
- [Bindings](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L87)
- [IncrementCounter](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L97)
- ...